pystac/extensions/raster.py

```diff
@@ -0,0 +1,715 @@
+    def statistics(self, v: Optional[Statistics]) -> None:
+        if v is not None:
+            self.properties["statistics"] = v.to_dict()
+        else:
+            self.properties.pop("statistics", None)
+
+    @property
+    def unit(self) -> Optional[str]:
+        """Get or sets the unit denomination of the pixel value
+
+        Returns:
+            [str]
+        """
+        return self.properties.get("unit")
+
+    @unit.setter
+    def unit(self, v: Optional[str]) -> None:
+        if v is not None:
+            self.properties["unit"] = v
+        else:
+            self.properties.pop("unit", None)
+
+    @property
+    def scale(self) -> Optional[float]:
+        """Get or sets the multiplicator factor of the pixel value to transform
+        into the value (i.e. translate digital number to reflectance).
+
+        Returns:
+            [float]
+        """
+        return self.properties.get("scale")
+
+    @scale.setter
+    def scale(self, v: Optional[float]) -> None:
+        if v is not None:
+            self.properties["scale"] = v
+        else:
+            self.properties.pop("scale", None)
+
+    @property
+    def offset(self) -> Optional[float]:
+        """Get or sets the number to be added to the pixel value (after scaling)
+        to transform into the value (i.e. translate digital number to reflectance).
+
+        Returns:
+            [float]
+        """
+        return self.properties.get("offset")
+
+    @offset.setter
+    def offset(self, v: Optional[float]) -> None:
+        if v is not None:
+            self.properties["offset"] = v
+        else:
+            self.properties.pop("offset", None)
+
+    @property
+    def histogram(self) -> Optional[Histogram]:
+        """Get or sets the Histogram distribution information of the pixels values in the band
+
+        Returns:
+            [Histogram]
+        """
+        return Histogram.from_dict(get_opt(self.properties.get("histogram")))
+
+    @histogram.setter
+    def histogram(self, v: Optional[Histogram]) -> None:
+        if v is not None:
+            self.properties["histogram"] = v.to_dict()
+        else:
+            self.properties.pop("histogram", None)
+
+    def __repr__(self) -> str:
+        return "<Raster Band>"
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Returns the dictionary representing the JSON of this Band.
+
+        Returns:
+            dict: The wrapped dict of the Band that can be written out as JSON.
+        """
+        return self.properties
+
+
+class RasterExtension(
+    Generic[T], PropertiesExtension, ExtensionManagementMixin[pystac.Item]
+):
+    """RasterItemExt is the extension of the Item in the raster extension
+
+    Args:
+        item : The item to be extended.
+
+    Attributes:
+        item : The Item that is being extended.
+
+    Note:
+        Using RasterItemExt to directly wrap an item will add
+        the 'raster' extension ID to the item's stac_extensions.
+    """
```

Comment:
Sorry, I've been doing some work in other PRs to update these docstrings and I think this may have been copied over from an earlier version.

```suggestion
    """An abstract class that can be used to extend the properties of an
    :class:`~pystac.Item` or :class:`~pystac.Asset` with properties from 
    the :stac-ext:`Raster Extension <raster>`. This class is generic over 
    the type of STAC Object to be extended (e.g. :class:`~pystac.Item`, 
    :class:`~pystac.Asset`).

    This class will generally not be used directly. Instead, use the concrete
    implementation associated with the STAC Object you want to extend (e.g.
    :class:`~ItemRasterExtension` to extend an :class:`~pystac.Item`).
    """
```